pkg/storage/storage_test.go

```diff
@@ -0,0 +1,327 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestStorage_AddProfile(t *testing.T) {
+	tests := []struct {
+		name       string
+		storage    *Storage
+		newProfile *Profile
+		validate   func(*testing.T, *Storage)
+	}{
+		{
+			name: "add new Profile",
+			storage: &Storage{
+				Default: "default",
+				Profiles: []*Profile{
+					{Name: "default", Key: "key"},
+				},
+			},
+			newProfile: &Profile{Name: "test", Key: "key"},
+			validate: func(t *testing.T, s *Storage) {
```

Comment:
I personally prefer to keep tests matrix declarative as much as possible so it should not contain any imperative constructions. I prefer this because imperative approach on tests matrix ruins the idea of that approach since it's mainly about keeping the same tests in the only table and their way to validate should also be the same.